cat-launcher/src/pages/SettingsPage/hooks/useSettings.ts

```diff
@@ -0,0 +1,206 @@
+    if (settingsError && onSettingsErrorRef.current) {
+      onSettingsErrorRef.current(settingsErrorObj as Error);
+    }
+  }, [settingsError, settingsErrorObj]);
+
+  useEffect(() => {
+    if (fontsError && onFontsErrorRef.current) {
+      onFontsErrorRef.current(fontsErrorObj as Error);
+    }
+  }, [fontsError, fontsErrorObj]);
+
+  useEffect(() => {
+    if (themesError && onThemesErrorRef.current) {
+      onThemesErrorRef.current(themesErrorObj as Error);
+    }
+  }, [themesError, themesErrorObj]);
+
+  const resolvedFonts = fonts ?? [];
+  const resolvedThemes = themes ?? [];
+
+  let themeName = "Custom";
+  if (settings?.theme && resolvedThemes.length > 0) {
+    const currentTheme = settings.theme;
+    const foundTheme = resolvedThemes.find((t) => {
+      return isThemeEqual(t.colors, currentTheme);
+    });
+    if (foundTheme) {
+      themeName = foundTheme.name;
+    }
+  }
+
+  const combinedSettings = settings
+    ? {
+        ...settings,
+        themeName,
+      }
+    : undefined;
+
+  return {
+    settings: combinedSettings,
+    fonts: resolvedFonts,
+    themes: resolvedThemes,
+    isLoading,
+  };
+}
+
+export function useFonts(options?: {
+  onError?: (err: Error) => void;
+}) {
+  const onErrorRef = useRef(options?.onError);
+
+  useEffect(() => {
+    onErrorRef.current = options?.onError;
+  });
+
+  const {
+    data: fonts,
+    isLoading,
+    isError,
+    error,
+  } = useQuery({
+    queryKey: queryKeys.fonts(),
+    queryFn: async () => {
+      return await listFonts();
+    },
+  });
+
+  useEffect(() => {
+    if (isError && onErrorRef.current) {
+      onErrorRef.current(error as Error);
+    }
+  }, [isError, error]);
+
+  return { fonts: fonts ?? [], isLoading };
+}
+
+export function useThemes(options?: {
+  onError?: (err: Error) => void;
+}) {
+  const onErrorRef = useRef(options?.onError);
+
+  useEffect(() => {
+    onErrorRef.current = options?.onError;
+  });
+
+  const {
+    data: themes,
+    isLoading,
+    isError,
+    error,
+  } = useQuery({
+    queryKey: queryKeys.themes(),
+    queryFn: async () => {
+      return await listThemes();
+    },
+  });
+
+  useEffect(() => {
+    if (isError && onErrorRef.current) {
+      onErrorRef.current(error as Error);
+    }
+  }, [isError, error]);
+
+  return { themes: themes ?? [], isLoading };
+}
+
+export function useApplySettings(options?: {
+  onError?: (err: Error) => void;
+}) {
+  const queryClient = useQueryClient();
+
+  const { mutateAsync: handleApplySettings, isPending } = useMutation(
+    {
+      mutationFn: async (params: {
+        maxBackups: number;
+        parallelRequests: number;
+        fontLocation?: string;
+        themeColors?: ThemeColors;
+      }) => {
+        await applySettings(params);
+      },
+      onSuccess: () => {
+        queryClient.invalidateQueries({
+          queryKey: queryKeys.settings(),
+        });
+      },
+      onError: (err) => {
+        options?.onError?.(err as Error);
```

Comment:
Pass onError directly. Don't pass it through the options object. Make this change in all places where options object is used.